sdk/servicebus/perf-tests/service-bus/test/receiveBatch.spec.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ServiceBusReceiver, ServiceBusSender } from "@azure/service-bus";
+import { PerfOptionDictionary } from "@azure/test-utils-perf";
+import { ServiceBusTest } from "./sbBase.spec";
+
+interface ReceiverOptions {
+  "max-message-count": number;
+  "number-of-messages": number;
+  "message-body-size-in-bytes": number;
+}
+
+export class BatchReceiveTest extends ServiceBusTest<ReceiverOptions> {
```

Comment:
I think `BatchReceiveTest` should extend `BatchPerfTest` rather than `PerfTest` (which `ServiceBusTest` extends).

While it is possible in JS to extend `PerfTest` and then override the `runBatch()` method, it should not be recommended, and this is not possible in other languages like .NET, where `PerfTest` seals the `runBatch()` method to prevent further overriding:

https://github.com/Azure/azure-sdk-for-net/pull/24653/files#diff-427d3566282e8c258aa70a4b5d4eddbbfc5cf250b38a8e0dcfcb7b43126b9f18R15

TypeScript does not support "sealed/final" methods, so I'm not sure if there is any reasonable way to prevent overriding `PerfTest.runBatch()`, but we should at least discourage it and make sure we don't do it in our examples.

Reply:
So, I'd either create two base classes like `ServiceBusTest extends PerfTest` and `ServiceBusBatchTest extends BatchPerfTest`, or just use `ServiceBusTest extends BatchPerfTest`, and have all tests override `runBatch()` (and return `1` if only performing a single operation).

Reply:
> ServiceBusTest extends BatchPerfTest, and have all tests override runBatch() (and return 1 if only performing a single operation)

My intention was to reuse the setup hence updated PerfTest to have both.
I like your suggested approach, will update.

Reply:
- We don't have "sealed"
- We could try decorators, but they are experimental yet and subject to change
- We have "abstract", "public", "private", "protected" and "override" keywords
   - We use "abstract" as intended
   - We can't use "private"/"protected" for run/runBatch
   - Override is not relevant
   - Looks like we need to make run and runBatch public
- However, I updated the base SBTestBase to extend the BatchPerfTest instead.